polaris-core/src/main/java/org/apache/polaris/core/auth/PolarisAuthorizerImpl.java

```diff
@@ -460,10 +462,13 @@ public class PolarisAuthorizerImpl implements PolarisAuthorizer {
   }
 
   private final PolarisConfigurationStore featureConfig;
+  private final Provider<PolarisGrantManager> grantManagerFactory;
 
   @Inject
-  public PolarisAuthorizerImpl(PolarisConfigurationStore featureConfig) {
+  public PolarisAuthorizerImpl(
+      PolarisConfigurationStore featureConfig, Provider<PolarisGrantManager> grantManagerFactory) {
```

Comment:
Why are we injecting a `Provider` instead of `PolarisGrantManager` directly?

Reply:
The `PolarisAuthorizer` is in a different scope from the `PolarisGrantManager`. Since the authorizer isn't realm or request-scoped, I prefer fetching the realm-scoped bean when it's needed. I know I can rely on runtime proxies, but... I dislike runtime proxies when they're not needed :)

Reply:
`PolarisAuthorizerImpl` is a light-weight object. Does it hurt to make it request-scoped? Cross-scope references resolved at call time are indeed to be avoided, but, IMHO, the `Provider` falls into the same category.